node/light_test.go

```diff
@@ -0,0 +1,26 @@
+package node
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/node/config"
+)
+
+func TestNewLight(t *testing.T) {
+	startCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	nd, err := NewLight(startCtx, &config.Config{})
+	assert.NoError(t, err)
+	require.NotNil(t, nd)
```

Comment:
That's more like a question more than a nit. The difference between `require` and `assert` is that the former interrupts the test execution with Fatal, while the latter is not. Usually, I use `assert` for things I specifically __testing__ and `require` for code that a test assumes to be functional. 

For example, I test some IO program that needs a TCP connection. When I create a conn I receive an error and for this case in a test, I would use `require`, because the program can't be tested without a valid connection, but for my code, I would use assert to test that it works, without interruption.

In the case above it does not make a lot of sense to do assert and require in that order, but by using the rule above I've just highlighted what I test specifically. 

We can reconsider the use of this rule in this repo though.

Reply:
Okay I'm fine with either.